cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/CassandraCradleStorage.java

```diff
@@ -595,19 +587,23 @@ public TestEventsMessagesLinker getTestEventsMessagesLinker()
 		LocalTime fromTime = fromDateTime.toLocalTime(),
 				toTime = toDateTime.toLocalTime();
 		
-		CompletableFuture<MappedAsyncPagingIterable<RootTestEventEntity>> future = new AsyncOperator<MappedAsyncPagingIterable<RootTestEventEntity>>(semaphore)
-				.getFuture(() -> ops.getRootTestEventOperator().getTestEvents(instanceUuid, fromDateTime.toLocalDate(), fromTime, toTime, readAttrs));
-		return future.thenApply(it -> new RootTestEventsMetadataIteratorAdapter(it));
+		RootTestEventOperator op = ops.getRootTestEventOperator();
+		CompletableFuture<MappedAsyncPagingIterable<RootTestEventEntity>> future = 
+				new AsyncOperator<MappedAsyncPagingIterable<RootTestEventEntity>>(semaphore)
+						.getFuture(() -> order == Order.DIRECT
+								? op.getTestEventsDirect(instanceUuid, fromDateTime.toLocalDate(), fromTime, toTime, readAttrs)
+								: op.getTestEventsInverse(instanceUuid, fromDateTime.toLocalDate(), fromTime, toTime, readAttrs));

```

Comment:
"Reversed" is more widely used in programming than "Inverse", if you ask me. "Inverse" is more "mathematical".